### PR TITLE
fix: Remove redact-pii dependency and address security vulnerabilities

### PR DESCRIPTION
**Pull request by naji247**
1. Security Vulnerability Fixes (package.json)

  Added pnpm.overrides to fix transitive dependency vulnerabilities:

  | Package                | Vulnerability            | Source Dependency         |
  |------------------------|--------------------------|---------------------------|
  | js-yaml ≥4.1.1         | Prototype pollution      | @changesets/cli           |
  | tmp ≥0.2.4             | Symlink attack           | @changesets/cli           |
  | vite ≥6.4.1            | Multiple security issues | vitest                    |
  | body-parser ≥2.2.1     | DoS via URL encoding     | @modelcontextprotocol/sdk |
  | brace-expansion =2.0.2 | ReDoS vulnerability      | eslint                    |

  ---
  2. Removed Runtime Dependencies

  Before: zod and redact-pii were runtime dependencies
  After:
  - zod moved to devDependencies (only needed for tests)
  - redact-pii completely removed (had outdated dependencies)

  This reduces the package footprint and eliminates security risks from redact-pii.

  ---
  3. Removed zod-compat.ts Module (185 lines deleted)

  The entire Zod v3/v4 compatibility layer was removed. The SDK no longer manipulates Zod schemas directly.

  ---
  4. Refactored Context Parameter Injection (context-parameters.ts)

  Key change: Context parameters are now injected into JSON Schema (after MCP SDK converts Zod → JSON Schema), not into Zod schemas.

  New safety checks added:
  - Skips tools that already have a context parameter (logs warning)
  - Skips complex schemas (oneOf/allOf/anyOf) that can't safely be extended
  - Removes additionalProperties: false constraint when adding context (required for schema validity)

  ---
  5. Moved Context Injection Timing (tracingV2.ts → tracing.ts)

  Context parameter injection now happens in setupListToolsTracing() instead of during tool registration:

  tracing.ts:74-87  // Inject context parameters AFTER MCP SDK conversion

  This ensures we work with JSON Schema, not raw Zod objects.

  ---
  6. Inlined Minimal Zod Helpers (tracingV2.ts)

  Kept only the essential helpers (getObjectShape, getLiteralValue) needed for extracting method names from MCP SDK internal schemas. These don't require importing Zod.

  ---
  7. Centralized get_more_tools Descriptor (tools.ts)

  Added:
  - GET_MORE_TOOLS_NAME constant
  - getReportMissingToolDescriptor() function returning JSON Schema (not Zod)

  Used consistently across the codebase to avoid duplication.

  ---
  8. Test File Changes

  - Deleted: context-parameters-zod.test.ts (155 lines) — tested Zod schema manipulation
  - Deleted: zod-compat.test.ts (276 lines) — tested the removed compat layer
  - Added: context-parameters-edge-cases.test.ts (232 lines) — tests new JSON Schema handling
  - Updated: context-parameters.test.ts — adjusted for new behavior

  ---
  9. Version Bump

  0.1.9 → 0.1.10

  ---
  Summary Table

  | Area         | Change                                   |
  |--------------|------------------------------------------|
  | Security     | 5 vulnerability overrides added          |
  | Dependencies | Removed redact-pii, moved zod to devDeps |
  | Architecture | JSON Schema manipulation instead of Zod  |
  | Code removed | ~660 lines (zod-compat + related tests)  |
  | Code added   | ~430 lines (edge case handling + tests)  |

### CODE REVIEW

File: src/modules/tracingV2.ts [hunk 1]
```diff
@@ -28,29 +26,101 @@ function isToolResultError(result: any): boolean {
...
+}
+
+interface ZodV4Internal {
+  _zod?: {
+    def?: {
+      value?: unknown;
+      values?: unknown[]; // For enums - some Zod versions store literal values here
+      shape?: Record<string, unknown> | (() => Record<string, unknown>);
+    };
+  };
```
[the remark this answers is not in the document]
Comment (kashishhora) [reply src/modules/tracingV2.ts]: nor the `getObjectShape` and `getLiteralValue`?

File: src/modules/tracingV2.ts [hunk 2]
```diff
@@ -176,13 +235,26 @@ function setupListenerToRegisteredTools(server: HighLevelMCPServerLike): void {
   }
 }
 
-function addMCPcatToolsToServer(server: HighLevelMCPServerLike): void {
+function _addMCPcatToolsToServer(server: HighLevelMCPServerLike): void {
```
Comment (kashishhora) [inline src/modules/tracingV2.ts]: seems like this is unused - remove?

File: src/modules/logging.ts [hunk 3]
```diff
@@ -42,7 +45,8 @@ export function writeToLog(message: string): void {
   const timestamp = new Date().toISOString();
   const logEntry = `[${timestamp}] ${message}`;
 
-  // Edge environment: use console.log as fallback
+  // Edge environment: use console.error as fallback (stderr, not stdout)
+  // This is critical for MCP servers using stdio transport
   if (useConsoleFallback) {
     console.log(`[mcpcat] ${logEntry}`);
```
Comment (kashishhora) [inline src/modules/logging.ts:49]: should be console.error right?

**Review by kashishhora — APPROVED** [review]

File: src/modules/logging.ts (hunk 3 above)
Comment (naji247) [reply src/modules/logging.ts:49]: Oh I changed to log, let me delete that
